### PR TITLE
Re-add action prototypes

### PR DESCRIPTION
**Pull request by ElectroJr**
- Re-adds action prototypes
- Makes the scream, combat-toggle, disarm, and the light-toggle actions use prototypes
- Re-names and adds more docstrings to some events
  - `GetActionsEvent` is now `GetItemActionsEvent` to make it clearer that it is intended for items/clothing related actions.
  - `PerformActionEvent` is now `BaseActionEvent`
  - `InstantActionEvent` is a new event that inherits from `BaseActionEvent`. It adds nothing new to the parent class, it just helps avoid confusion.
  - `PerformEntityTargetActionEvent` is now just `EntityTargetActionEvent`
  - `PerformWorldTargetActionEvent` is now just `WorldTargetActionEvent`

The way that action prototypes have to be cloned is a bit shit, and would be fixed if the action-state data (cooldown, charges, toggled-status) were moved out of the action class into some separate `DataDefinition`.

### CODE REVIEW

File: Content.Shared/Actions/ActionTypes/ActionPrototypes.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+public sealed class WorldTargetActionPrototype : WorldTargetAction, IPrototype
+{
+    [IdDataFieldAttribute]
+    public string ID { get; } = default!;
+
+    // This is a shitty hack to get around the fact that action-prototypes should not in general be sever-exclusive
+    // prototypes, but some actions may need to use server-exclusive events, and there is no way to specify on a
+    // per-prototype basis whether the client should ignore it when validating yaml.
+    [DataField("serverEvent", serverOnly: true)]
+    public WorldTargetActionEvent? SeverEvent
```
Comment (metalgearsloth) [inline Content.Shared/Actions/ActionTypes/ActionPrototypes.cs:22]: server

**Review by metalgearsloth — APPROVED** [review]
The ID / action split is kinda uggo but not sure of a better solution atm.

**Review by DrSmugleaf — APPROVED** [review]

File: Content.Shared/Actions/ActionTypes/ActionPrototypes.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Content.Shared/Actions/ActionTypes/ActionPrototypes.cs]: Fixed in [`fb22b64` (#7508)](https://github.com/space-wizards/space-station-14/pull/7508/commits/fb22b648964b4f7e305a1306f835d880c7459ab6)

**Comment by DrSmugleaf** [comment]
This is good for the HUD refactor

**Pull request merged**